extensions/mypy_extensions.py

```diff
@@ -8,15 +8,65 @@
 # NOTE: This module must support Python 2.7 in addition to Python 3.x
 
 
-def TypedDict(typename, fields):
-    """TypedDict creates a dictionary type that expects all of its
+import sys
+from typing import _type_check  # type: ignore
+
+
+def _check_fails(cls, other):
+    if sys._getframe(1).f_globals['__name__'] not in ['abc', 'functools']:
```

Comment:
(1) Is there a reason that these particular modules are special-cased so that they *do* allow isinstance checks?

(2) Perhaps the call to sys._getframe() should be guarded when using a non-CPython implementation that doesn't support sys._getframe()? I see this is done in `_TypedDictMeta.__new__` below.